### PR TITLE
Remove packages/{management,standalone} dirs and move contents to parent dir

### LINKED ISSUE #1782: Remove packages/{management,standalone}/ directories and move their contents to packages/ directory
(This is used to request new product features)

**Describe the feature request**

Remove `packages/{management,standalone}` directories and move their contents to the parent, `packages` directory. The following will break with the change and will need to be updated to work with the new structure:

- Makefile
- hack/packages/scripts/create-package.sh
- hack/packages/package-tools/cmd/package-bundle-generate.go
- hack/packages/package-tools/cmd/package-bundle-push.go
- hack/packages/package-tools/cmd/repo-bundle-generate.go
- hack/packages/package-tools/cmd/repo-bundle-push.go
- hack/packages/package-tools/cmd/vendir-sync.go

**Describe alternatives you've considered**

None

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

Carvel users expect package bundles to be in the packages directory, and not in an intermediary directory.

Currently, the structure for the `featuregates` package bundle is `packages/management/featuregates/bundle/`. However, this is unexpected. The expectation is `packages/featuregates/bundle/`. This is the reason for the directory structure change.

To avoid git merge conflicts, I'm going to wait for PR #1512  to be merged before I create a PR to fix this issue.

/cc: @yharish991

### PR DESCRIPTION
**Pull request by codegold79**
### What this PR does / why we need it

Remove packages/{management,standalone} directories and move their contents to the packages directory. 

We need this because Carvel packages are expected to be in the `packages` directory and not a subdirectory of `packages`.

**Other Notables**
- I did update `pkg/v1/providers/tests/unit/fixtures`. Should the `management` be removed from `imagePath:sandbox/packages/management/repo`? I kept it in for now because it's still a valid unit test and does not affect production logic.
- I kept the repo directories `standalone` and `management` in `build/package-repo-bundles`. 
- Repo bundles now have the subversion in the tar.gz and Package CR file names (previously, only had version)
- For package bundles (but not for repo bundles because the Repository struct in `hack/packages/package-tools/cmd/packageValues.go` does not have a subVersion field), the `PackageSubVersion` is taken from `package-values.yaml` The `subVersion` flag overrides `packageSubVersion`.
- Package CR and tarballs also include the `PackageSubVersion` if provided and not overridden by the `subVersion` flag

### Which issue(s) this PR fixes

Fixes #1782

### Describe testing done for PR

I manually ran on my MacBook Pro the following`make` targets and ensured that packages built and got pushed to a container registry without problems. To repeat my test, make sure to `export OCI_REGISTRY` and set it to a registry you have read and write access to. Here are some of the commands I was able to run without error:
   - make create-package PACKAGE_NAME=pr-test-pkg
   - PACKAGE_NAME=tkg make package-bundle 
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_NAME=standalone-plugins make package-bundle
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_NAME=capabilities make package-bundle
   - REPO_BUNDLE_SUB_VERSION="may.03.22" PACKAGE_REPOSITORY=standalone make package-repo-bundle  
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_BUNDLES=standalone-plugins,capabilities make push-package-bundles 
   - REPO_BUNDLE_SUB_VERSION="may03.22" make push-package-repo-bundle 
   - make package-vendir-sync
        - This test resulted in a correction to the `- path: webhook-manifests.lib.yaml` in `packages/addons-manager-v2/vendir.yml#L58`. /cc @vijaykatam 
   - PACKAGE_SUB_VERSION="may.03.22" PACKAGE_REPOSITORY=standalone make package-push-bundles-repo

@tylerschultz  for pointed out this test
   - cd hack/provider-sync-tools && make validate

Here is what the new directory structure looks like in the `build` directory:
build-directory-structure

### Release note

```release-note
    Removed packages/{management,standalone} directories and move their contents to the packages directory. 
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: hack/packages/package-tools/cmd/package-bundle-generate.go
Comment (yharish991) [inline hack/packages/package-tools/cmd/package-bundle-generate.go:33]: i would still like to keep `packageRepository`, but make it optional.. that way if someone is working just on that package repository they can build and play with packages in that repo

File: hack/packages/package-tools/cmd/package-bundle-push.go
Comment (yharish991) [inline hack/packages/package-tools/cmd/package-bundle-push.go:30]: same as above, it would keep it optional

File: hack/packages/package-tools/cmd/vendir-sync.go
Comment (yharish991) [inline hack/packages/package-tools/cmd/vendir-sync.go:28]: same as above, i would like to keep it optional

**Review by yharish991 — CHANGES_REQUESTED** [review]

File: hack/packages/package-tools/cmd/vendir-sync.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/vendir-sync.go:28]: This one is slightly more difficult than `package-bundle-generate` and `push-package-bundle`. Currently, the way vendir-sync was run loop through files in a particular repo directory, if repository was specified with `PACKAGE_REPOSITORY`. Here is the code:

```go
for _, file := range files {
		if file.IsDir() {
			fmt.Printf("Syncing package %s\n", file.Name())
			packagePath := filepath.Join(repositoryPath, file.Name())
			if err := syncPackage(packagePath, toolsBinDir); err != nil {
				return err
			}
		}
	}
```

Now that the repository directories are gone, I'll need to figure out which packages are in the requested repository. 

What I plan to do is load the `packages/package-values.yaml file`, collect the package names in the provided repository, then do vendir sync for those packages.

File: hack/packages/package-tools/cmd/package-bundle-generate.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/package-bundle-generate.go]: Good call. I have made PACKAGE_REPOSITORY an option now.

File: hack/packages/package-tools/cmd/package-bundle-push.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/package-bundle-push.go]: I agree. It's an option now.

File: hack/packages/package-tools/cmd/vendir-sync.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/vendir-sync.go]: Done

**Comment by codegold79** [comment]
I am keeping this PR as a WIP until #1238 is merged as I want to update packaging documentation.

File: hack/packages/package-tools/cmd/package-bundle-push.go
Comment (codegold79) [inline hack/packages/package-tools/cmd/package-bundle-push.go:123]: rename args to be more explanatory

File: hack/packages/package-tools/cmd/package-bundle-push.go
Comment (codegold79) [inline hack/packages/package-tools/cmd/package-bundle-push.go:59]: Pass in first argument only. Do len(args) check before this line.

File: hack/packages/package-tools/cmd/vendir-sync.go
Comment (codegold79) [inline hack/packages/package-tools/cmd/vendir-sync.go:85]: Use filterPackageRepo instead

Comment (codegold79) [reply hack/packages/package-tools/cmd/vendir-sync.go]: @yharish991, I made use of `filterPackageRepos` and put the changes in a separate commit so you can see what had to change. I did indeed have to make another helper function to look through a slice and look for a particular package. Let me know if you want me to keep the change or revert.

**Review by yharish991 — CHANGES_REQUESTED** [review]
repo bundle generation is failing, can you remove `packageRepository` in this line https://github.com/vmware-tanzu/tanzu-framework/pull/1906/files#diff-6f3e1d6f0cf8e16f6f2980e8f86cbc15115703f046cb9d75d2632dc37e3ea16aR238

Also, can you rebase and fix conflicts

File: hack/packages/package-tools/cmd/package-bundle-push.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/package-bundle-push.go:136]: I don't think improving the performance (like you said, we're looping through just two repos) is worth the change, but perhaps readability is. However, I think it's not _that_ hard to read the nested for loops.

I'll let this one go (I won't refactor code to remove a for loop), mainly because at the moment, I'm not sure how even the new data structure you proposed would help. I could be wrong, but even with the new data structure, you need to know the repo name don't you? Not knowing the repo name, e.g. "management", then you'll still be forced to loop through all the repositories.

**Comment by codegold79** [comment]
>  repo bundle generation is failing, can you remove packageRepository in this line https://github.com/vmware-tanzu/tanzu-framework/pull/1906/files#diff-6f3e1d6f0cf8e16f6f2980e8f86cbc15115703f046cb9d75d2632dc37e3ea16aR238

Yes,  I was able to find `packageRespository` and remove it. The change can be seen in repo-bundle-generate.go line 240.

> Also, can you rebase and fix conflicts
Done.

@yharish991, I created new commit messages with your requested changes so it can be easier for you to see the diffs between reviews. I'll be sure to squash all commits into one before I merge to main.

File: docs/packages/add-new-package-to-package-repo.md
Comment (yharish991) [inline docs/packages/add-new-package-to-package-repo.md:15]: we can remove `PACKAGE_REPOSITORY` var here

**Review by yharish991 — COMMENTED** [review]
LGTM, added a minor comment suggesting a change in docs, will approve this pr after the comment is resolved

**Comment by yharish991** [comment]
Can you also update the testing done in the pr description with the make targets you used for testing?
Also, can you update the release note like you are in the future? https://github.com/vmware-tanzu/tanzu-framework/blob/main/docs/release/release-notes.md#contents-of-a-release-note

File: hack/packages/package-tools/cmd/vendir-sync.go
Comment (codegold79) [reply hack/packages/package-tools/cmd/vendir-sync.go]: I'll resolve this comment. I take from @yharish991 's later [comment](https://github.com/vmware-tanzu/tanzu-framework/pull/1906#pullrequestreview-939707125) that this latest version where `filterPackageRepos` is used is preferable and I don't need to revert to using `selectPackages`.

File: docs/packages/add-new-package-to-package-repo.md
Comment (codegold79) [reply docs/packages/add-new-package-to-package-repo.md]: Done

**Review by tylerschultz — APPROVED** [review]
LGTM

**Review by yharish991 — APPROVED** [review]

**Comment by rajathagasthya** [comment]
@anujc25 Can you review and confirm standalone packages look good?

**Review by yharish991 — APPROVED** [review]
LGTM!! did another round of review, thanks for making these changes @fgold

**Review by miclettej — APPROVED** [review]

**Pull request merged**
